eng/common/scripts/Update-DocsMsMetadata.ps1

```diff
@@ -196,16 +195,22 @@ function UpdateDocsMsMetadataForPackage($packageInfoJsonLocation, $packageInfo)
 
 foreach ($packageInfoLocation in $PackageInfoJsonLocations) {
   Write-Host "Updating metadata for package: $packageInfoLocation"
-
-  # Convert package metadata json file to metadata json property.
-  $packageInfo = GetPackageInfoJson $packageInfoLocation
-  # Add validation step for daily update and release
-  if ($ValidateDocsMsPackagesFn -and (Test-Path "Function:$ValidateDocsMsPackagesFn")) {
-    &$ValidateDocsMsPackagesFn -PackageInfo $packageInfo -PackageSourceOverride $PackageSourceOverride -DocValidationImageId $DocValidationImageId -DocRepoLocation $DocRepoLocation
-    if ($LASTEXITCODE) {
-      LogError "The package failed Doc.Ms validation. Check https://aka.ms/azsdk/docs/docker for more details on how to diagnose this issue."
-      exit $LASTEXITCODE
+  try {
+    # Convert package metadata json file to metadata json property.
+    $packageInfo = GetPackageInfoJson $packageInfoLocation
+    # Add validation step for daily update and release
+    if ($ValidateDocsMsPackagesFn -and (Test-Path "Function:$ValidateDocsMsPackagesFn")) {
+      Write-Host "Validating the package..."
+      &$ValidateDocsMsPackagesFn -PackageInfo $packageInfo -PackageSourceOverride $PackageSourceOverride -DocValidationImageId $DocValidationImageId -DocRepoLocation $DocRepoLocation
+      if ($LASTEXITCODE) {
+        LogError "The package failed Doc.Ms validation. Check https://aka.ms/azsdk/docs/docker for more details on how to diagnose this issue."
+        exit $LASTEXITCODE
+      }
     }
+    Write-Host "Updating the package json ..."
+    UpdateDocsMsMetadataForPackage $packageInfoLocation $packageInfo
+  }
+  catch {
+    Write-Error $_
```

Comment:
Is the error not already getting printed and stopping execution here? I'm not sure I understand why the catch block is necessary.